examples/goldilocks_models/initial_guess.cc

```diff
@@ -0,0 +1,195 @@
+            MatrixXd w_gamma;
+            VectorXd weight_gamma;
+            for (sample_num = 0; sample_num < total_sample_num; sample_num++) {
+//                check if this sample is success
+                int is_success = (readCSV(directory + to_string(past_iter) + string("_")
+                        + to_string(sample_num) + string("_is_success.csv")))(0,0);
+                if(is_success == 1) {
+                    MatrixXd past_ground_incline = readCSV(directory + to_string(past_iter) + string("_")
+                            + to_string(sample_num) + string("_ground_incline.csv"));
+                    MatrixXd past_stride_length = readCSV(directory + to_string(past_iter) + string("_")
+                            + to_string(sample_num) + string("_stride_length.csv"));
+                    MatrixXd past_turning_rate = readCSV(directory + to_string(past_iter) + string("_")
+                            + to_string(sample_num) + string("_turning_rate.csv"));
+                    VectorXd past_gamma(gamma_dimension);
+                    past_gamma << past_ground_incline(0, 0), past_stride_length(0, 0), past_turning_rate(0, 0);
+                    // normalize gamma
+                    VectorXd normalize_scale = VectorXd::Zero(gamma_dimension);
+                    if(! (min_gi==max_gi))
+                    {
+                        normalize_scale[0] = 1/(max_gi-min_gi);
+                    }
+                    if(! (min_sl==max_sl))
+                    {
+                        normalize_scale[1] = 1/(max_sl-min_sl);
+                    }
+                    if(! (min_tr==max_tr))
+                    {
+                        normalize_scale[2] = 1/(max_tr-min_tr);
+                    }
+                    VectorXd dif_gamma = (past_gamma - current_gamma).array().abs()*normalize_scale.array();
+                    VectorXd dif_gamma2 = dif_gamma.array().pow(2);
+                    double distance_gamma =  (dif_gamma.transpose() * gamma_scale * dif_gamma2)(0,0);
+                    VectorXd w_to_interpolate = readCSV(directory + to_string(past_iter) + string("_")
+                                                        + to_string(sample_num) + string("_w.csv"));
+                    if (distance_gamma == 0) {
+                        w_gamma.conservativeResize(w_to_interpolate.rows(), 1);
+                        w_gamma << w_to_interpolate;
+                        weight_gamma.conservativeResize(1);
+                        weight_gamma << 1;
+                        break;
+                    }
+                    else{
+                        w_gamma.conservativeResize(w_to_interpolate.rows(),w_gamma.cols()+1);
+                        w_gamma.col(w_gamma.cols()-1) = w_to_interpolate;
+                        weight_gamma.conservativeResize(weight_gamma.rows()+1);
+                        weight_gamma(weight_gamma.rows()-1)= 1 /distance_gamma;
+                    }
```

Comment:
I have a hard time understanding this part. Could you add more comments?

Reply:
And why are the codes between line 163 and line 175 identical to the ones here?

Reply:
Add more comments here.